backend/coreapp/admin.py

```diff
@@ -1,10 +1,32 @@
 from django.contrib import admin
 
-from .models import Profile, Assembly, Asm, Scratch
-from .github import GitHubUser
+import shutil
+
+from .models import Profile, Assembly, Asm, Scratch, Project, ProjectFunction
+from .github import GitHubUser, GitHubRepo
+
+class GitHubRepoAdmin(admin.ModelAdmin):
+    actions = ["pull", "reclone"]
+
+    @admin.action(description="Pull selected repos from GitHub")
+    def pull(self, request, queryset):
+        for repo in queryset.all():
+            repo.is_pulling = False
```

Comment:
Why false here and below in reclone?

Reply:
It forces a pull even if the database reckons its already doing one

Reply:
could you explain the scenario in which this might happen? I would imagine that if the database thinks there is one happening, we should leave it alone until it's done. 

On the other hand, if there *is* a pull happening in fact and we go to do it again, that could corrupt the state of the repo

Reply:
Good point